HPC/WRF.md

```diff
@@ -0,0 +1,263 @@
+```
+  * build netCDF-c
+```
+export INSTALLDIR=/shared
+export GCC_VERSION=10.2.0
+export OPENMPI_VERSION=4.1.4
+export PATH=${INSTALLDIR}/gcc-${GCC_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/gcc-${GCC_VERSION}/lib64:$LD_LIBRARY_PATH
+export PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/lib:$LD_LIBRARY_PATH
+export CC=mpicc
+export CXX=mpicxx
+export FC=mpif90
+export F77=mpif90
+export F90=mpif90
+HDF5=${INSTALLDIR}/hdf5
+PNET=${INSTALLDIR}/pnetcdf
+ZLIB=${INSTALLDIR}/zlib
+export CPPFLAGS="-I$HDF5/include -I${PNET}/include"
+export CFLAGS="-I$HDF5/include -I${PNET}/include"
+export CXXFLAGS="-I$HDF5/include -I${PNET}/include"
+export FCFLAGS="-I$HDF5/include -I${PNET}/include"
+export FFLAGS="-I$HDF5/include -I${PNET}/include"
+export LDFLAGS="-I$HDF5/include -I${PNET}/include -L$ZLIB/lib -L$HDF5/lib -L${PNET}/lib"
+cd /shared/tools
+wget -N https://downloads.unidata.ucar.edu/netcdf-c/4.8.1/netcdf-c-4.8.1.tar.gz
+tar -xzvf netcdf-c-4.8.1.tar.gz
+cd netcdf-c-4.8.1
+./configure --prefix=${INSTALLDIR}/netcdf CPPFLAGS="-I$HDF5/include -I$PNET/include" CFLAGS="-DHAVE_STRDUP -O3 -march=armv8.2-a+crypto+fp16+rcpc+dotprod" LDFLAGS="-L$HDF5/lib -L$PNET/lib" --enable-pnetcdf --enable-large-file-tests --enable-largefile  --enable-parallel-tests --enable-shared --enable-netcdf-4  --with-pic --disable-doxygen --disable-dap
+make -j$(nproc) && make install
+```
+  * install netCDF-fortran
+```
+export INSTALLDIR=/shared
+export GCC_VERSION=10.2.0
+export OPENMPI_VERSION=4.1.4
+export PATH=${INSTALLDIR}/gcc-${GCC_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/gcc-${GCC_VERSION}/lib64:$LD_LIBRARY_PATH
+export PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/lib:$LD_LIBRARY_PATH
+export CC=mpicc
+export CXX=mpicxx
+export FC=mpif90
+export F77=mpif90
+export F90=mpif90
+HDF5=${INSTALLDIR}/hdf5
+NCDIR=${INSTALLDIR}/netcdf
+export LD_LIBRARY_PATH=${NCDIR}/lib:${LD_LIBRARY_PATH}
+export CPPFLAGS="-I$HDF5/include -I$NCDIR/include"
+export CFLAGS="-I$HDF5/include -I$NCDIR/include"
+export CXXFLAGS="-I$HDF5/include -I$NCDIR/include"
+export FCFLAGS="-I$HDF5/include -I$NCDIR/include"
+export FFLAGS="-I$HDF5/include -I$NCDIR/include"
+export LDFLAGS="-L$HDF5/lib -L$NCDIR/lib"
+wget -N https://downloads.unidata.ucar.edu/netcdf-fortran/4.5.4/netcdf-fortran-4.5.4.tar.gz
+tar -xzvf netcdf-fortran-4.5.4.tar.gz
+cd netcdf-fortran-4.5.4
+./configure --prefix=$NCDIR --disable-static --enable-shared --with-pic --enable-parallel-tests --enable-large-file-tests --enable-largefile
+make -j$(nproc) && make install
+```
+  * Build WRF 4.2.2
+```
+git clone https://github.com/wrf-model/WRF.git
+git checkout release-v4.2.2
+```
+apply this [patch](WRF-v422-patch-correctness.diff)
+```
+git apply WRF-v422-patch-correctness.diff
+```
+
+  * Build WRF 4.5 - Fetch WRF code (release-v4.5), configure and compile
```

Comment:
it will be good to add some text saying, we recommend the release 4.5 or later to get then precision fixes, and then talk about older release patching